packages/core/base-component.ts

```diff
@@ -0,0 +1,10 @@
+export class BaseComponent extends HTMLElement {
+  protected init() {}
+  beforeRender() {}
+  afterRender() {}
+  connected() {}
+  disconnected() {}
+  render(root: ShadowRoot | HTMLDivElement) {}
+  /** @fixme is it never used? */
+  propChanged(name: string, oldValue: string, newValue: string) {}
```

Comment:
this callback it to be overridden in successor class when html attribute is changed

Reply:
In this case, please, describe how to use `propChanged` in repo `micro-frontends-starter`, because it looks like unreachable, dead code

Reply:
````es6
class extends Component {
    propChanged(name: string, oldValue: string, newValue: string) {
        // do smth here - re-render component, send request etc
    }
}
````

it's put here to be overridden in successor like any other callback (connected/disconnected)

Reply:
I meant for the framework users there must be a guide. And also for us, because we have some abstraction, which we never tried

Reply:
i agree - it should be added to the Component docs